### PR TITLE
fix: handling array of array in rest security accounts

### PR DESCRIPTION
**Pull request by Hardikl**
Parse applications and then created 1-1 mapping between application and methods. 
Now, ZAPI and REST would have same behaviour.

Fix: Handling this, need to call extra rest call in plugin.

image

### CODE REVIEW

File: cmd/collectors/rest/plugins/securityaccount/securityaccount.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		if securityAccountInstance := data.GetInstance(securityAccountKey); securityAccountInstance != nil {
+			securityAccountInstance.SetExportable(false)
+
+			for application, methods := range applicationToMethodMap {
+				for _, method := range methods {
+					var securityAccountNewInstance *matrix.Instance
+					securityAccountNewKey := securityAccountKey + application + method
+					if securityAccountNewInstance, err = s.data.NewInstance(securityAccountNewKey); err != nil {
+						s.Logger.Debug().Msgf("add instance failed for key %s: %v", securityAccountNewKey, err)
+						s.Logger.Error().Err(err).Str("Instance key", securityAccountNewKey).Msg("")
```
Comment (cgrinds) [inline cmd/collectors/rest/plugins/securityaccount/securityaccount.go:130]: the debug can be removed if the securityAccountNewKey is added to the error logging on the next line

Comment (Hardikl) [reply cmd/collectors/rest/plugins/securityaccount/securityaccount.go]: yes, removed.

**Review by cgrinds — APPROVED** [review]

File: cmd/collectors/rest/plugins/securityaccount/securityaccount.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	)
+
+	// Purge and reset data
+	s.data.PurgeInstances()
+	s.data.Reset()
+
+	// Set all global labels from Rest.go if already not exist
+	s.data.SetGlobalLabels(data.GetGlobalLabels())
+
+	href := rest.BuildHref("", "*", nil, "", "", "", "", s.query)
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/rest/plugins/securityaccount/securityaccount.go:91]: nope, will add filter here to reduce the load.

File: conf/rest/9.12.0/security_account.yaml [hunk 3]
```diff
@@ -3,17 +3,19 @@ name:             SecurityAccount
...
+  - data: 1m
+
 counters:
   - ^^name                                      => user_name
   - ^^owner.name                                => svm
   - ^role.name                                  => role_name
   - ^password_hash_algorithm                    => hash_algorithm
   - ^locked                                     => locked
-  - ^applications.#.application                 => applications
-  - ^applications.#.authentication_methods.0    => methods
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply conf/rest/9.12.0/security_account.yaml:13]: It's not always 1-1 mapping between application and methods.

        {
            "owner": {
                "uuid": "3cedfe6f-8b57-11e9-8974-00a098e0219a",
                "name": "ocum-infinity",
                "_links": {
                    "self": {
                        "href": "/api/svm/svms/3cedfe6f-8b57-11e9-8974-00a098e0219a"
                    }
                }
            },
            "name": "admin",
            "applications": [
                {
                    "application": "amqp",
                    "authentication_methods": [
                        "password"
                    ],
                    "second_authentication_method": "none",
                    "is_ldap_fastbind": false
                },
                {
                    "application": "console",
                    "authentication_methods": [
                        "password"
                    ],
                    "second_authentication_method": "none",
                    "is_ldap_fastbind": false
                },
                {
                    "application": "http",
                    "authentication_methods": [
                        "certificate",
                        "password"
                    ],
                    "second_authentication_method": "none",
                    "is_ldap_fastbind": false
                },
                {
                    "application": "ontapi",
                    "authentication_methods": [
                        "certificate",
                        "password"
                    ],
                    "second_authentication_method": "none",
                    "is_ldap_fastbind": false
                },
                {
                    "application": "service_processor",
                    "authentication_methods": [
                        "password"
                    ],
                    "second_authentication_method": "none",
                    "is_ldap_fastbind": false
                },
                {
                    "application": "ssh",
                    "authentication_methods": [
                        "password"
                    ],
                    "second_authentication_method": "none",
                    "is_ldap_fastbind": false
                }
            ],
            "role": {
                "name": "admin",
                "_links": {
                    "self": {
                        "href": "/api/svms/3cedfe6f-8b57-11e9-8974-00a098e0219a/admin/roles/admin"
                    }
                }
            },
            "locked": false,
            "password_hash_algorithm": "sha512",
            "scope": "cluster",
            "_links": {
                "self": {
                    "href": "/api/security/accounts/3cedfe6f-8b57-11e9-8974-00a098e0219a/admin"
                }
            }
        },



Here, applications are 6 but we need 8 records as 1 application(http and ontapi) can use multiple methods for authentication.

File: cmd/collectors/rest/plugins/securityaccount/securityaccount.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/rest/plugins/securityaccount/securityaccount.go]: done

**Pull request merged**